tests/func-tests/node_placement_test.go

```diff
@@ -17,39 +24,111 @@ import (
 	kvtutil "kubevirt.io/kubevirt/tests/util"
 
 	networkaddonsv1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1"
-	"kubevirt.io/client-go/kubecli"
-	"kubevirt.io/kubevirt/tests/flags"
-
 	tests "github.com/kubevirt/hyperconverged-cluster-operator/tests/func-tests"
+	"kubevirt.io/client-go/kubecli"
 )
 
-const nodePlacementSamplingSize = 20
-
-var _ = Describe("[rfe_id:4356][crit:medium][vendor:cnv-qe@redhat.com][level:system]Node Placement", func() {
+const (
+	hcoLabel    = "node.kubernetes.io/hco-test-node-type"
+	infra       = "infra"
+	workloads   = "workloads"
+	hcoResource = "hyperconvergeds"
+)
 
+var _ = Describe("[rfe_id:4356][crit:medium][vendor:cnv-qe@redhat.com][level:system]Node Placement", Ordered, func() {
 	var workloadsNode *v1.Node
-
+	ctx := context.TODO()
 	tests.FlagParse()
 	client, err := kubecli.GetKubevirtClient()
 	kvtutil.PanicOnError(err)
 
-	workloadsNodes, err := client.CoreV1().Nodes().List(context.TODO(), k8smetav1.ListOptions{
-		LabelSelector: "node.kubernetes.io/hco-test-node-type==workloads",
+	BeforeAll(func() {
+		nodes, err := client.CoreV1().Nodes().List(context.TODO(), k8smetav1.ListOptions{LabelSelector: "node-role.kubernetes.io/master!="})
+		kvtutil.PanicOnError(err)
+		//TODO(dharmit): add check to possibly skip node placement tests
```

Comment:
@nunnatsa where do you recommend we add a check for number of workload nodes, which helps us deduce if we should skip the test? Presently, it's done this way - https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/tests/func-tests/node_placement_test.go#L36-L54.

I think you mentioned we should skip if there's only one node. So maybe a check for `len(nodes.Items)==1` or `len(nodes.Items)<2`?

Reply:
I think we should skip it as soon as possible. so here is a good place.

just remember to check `node-role.kubernetes.io/control-plane` instead of `master`.